test/stdlib/RangeDiagnostics.swift

```diff
@@ -283,7 +282,7 @@ func disallowSubscriptingOnIntegers() {
     r3[0]       // expected-error {{cannot subscript a value of type 'CountableClosedRange<UInt>' with an index of type 'Int'}}
     // expected-note@-1 {{overloads for 'subscript'}}
 
-    r0[UInt(0)] // expected-error {{ambiguous reference to member 'subscript'}}
+    r0[UInt(0)] // expected-error {{cannot subscript a value of type 'CountableRange<Int>' with an index of type 'UInt'}}
```

Comment:
Ah yes, this is done on purpose (apparently) because in case of L278 there is no type problem in the expression - `CountableRange<Int>` and subscript argument is `Int`, but according to http://swiftdoc.org/nightly/type/CountableRange/ - it's explicitly forbidden to subscript Int range with Int because the indexing of ranges is not the same as indexing for arrays and other things. Where in case of L285 - we are trying to subscript `CountableRange<Int>` with `UInt` argument which is incorrect.